src/app/Http/Controllers/V1/DocumentDistributionController.php

```diff
@@ -0,0 +1,347 @@
+        $inbox->Tgl                 = Carbon::parse($this->document->tanggal)->format('Y-m-d H:i:s');
+        $inbox->JenisId             = $this->findDocumentTypeId();
+        $inbox->UrgensiId           = $this->findDocumentUrgency();
+        $inbox->SifatId             = $this->findDocumentConfidentiality();
+        $inbox->Nomor               = $this->document->nomor_akhir;
+        $inbox->Hal                 = $this->document->perihal;
+        $inbox->Pengirim            = 'internal';
+        $inbox->NTipe               = $this->findDocumentDraftType();
+        $inbox->Namapengirim        = $this->creator?->role->RoleDesc;
+        $inbox->NFileDir            = 'naskah';
+        $inbox->BerkasId            = '1';
+        $inbox->ClId                = $this->findClassificationId();
+        $inbox->sidebar_v2_tid      = $this->document->id;
+        $inbox->sidebar_v2_metadata = json_encode($this->document);
+        $inbox->tandatangan         = json_encode($this->document->tandatangan);
+        $inbox->save();
+    }
+
+    /**
+     * saveToInboxReceiver
+     * Save document to inbox_receiver table based on payload document v2
+     *
+     * @return void
+     */
+    protected function saveToInboxReceiver()
+    {
+        $InboxReceiver = new InboxReceiver();
+        $InboxReceiver->NId           = $this->inboxId;
+        $InboxReceiver->NKey          = TableSetting::first()->tb_key;
+        $InboxReceiver->GIR_Id        = $this->inboxId;
+        $InboxReceiver->From_Id       = $this->creator?->PeopleId;
+        $InboxReceiver->RoleId_From   = $this->creator?->PrimaryRoleId;
+        $InboxReceiver->To_Id         = $this->receiver?->PeopleId;
+        $InboxReceiver->RoleId_To     = $this->receiver?->PrimaryRoleId;
+        $InboxReceiver->ReceiverAs    = 'to_forward';
+        $InboxReceiver->StatusReceive = 'unread';
+        $InboxReceiver->ReceiveDate   = Carbon::now();
+        $InboxReceiver->To_Id_Desc    = $this->receiver?->role->RoleDesc;
+        $InboxReceiver->Status        = '0';
+        $InboxReceiver->action_label  = ActionLabelTypeEnum::REVIEW();
+        $InboxReceiver->save();
+    }
+
+    /**
+     * doSaveToInboxFile
+     * Transfer file to webhook v1 and save inbox files
+     *
+     * @return void
+     */
+    protected function doSaveToInboxFile()
+    {
+        // Setup pdf file name
+        $type = str_replace(' ', '_', ucwords($this->document->jenis));
+        $title = str_replace(' ', '_', trim(preg_replace('/[^a-zA-Z0-9_ -]/s', '', substr($this->document->perihal, 0, 180))));
+        $time = parseDateTimeFormat(Carbon::now(), 'dmY') . '_' . parseDateTimeFormat(Carbon::now(), 'His');
+        $pdfName = $type  . '_' . $title . '_' . $time  . '_signed.pdf';
+
+        // Save file from URL
+        Storage::disk('local')->put($pdfName, file_get_contents($this->document->file['path']));
+        $pdfFile = file_get_contents(Storage::path($pdfName), 'r');
```

Comment:
couldn't we just do this?

```suggestion
        $pdfFile = file_get_contents(file_get_contents($this->document->file['path']));
```

Reply:
or you mean we didn't need to save file and use file upload with `file_get_contents` return, CMIIW

Reply:
this @indraprasetya154 

> or you mean we didn't need to save file and use file upload with file_get_contents return, CMIIW

from my understanding, the code here only need to download the file to be passed on to sikd-website. so I think it didn't actually need to be stored on hard drive. cmiiw

Reply:
I tried on my local and your suggestion is running well, thanks mas
change here [cded7c0](https://github.com/jabardigitalservice/office-services/pull/660/commits/cded7c0f22454415120f96b5a9787a12d7bf0228)